### ISSUE TITLE
Using multiple guards for a route

### ISSUE THREAD

**Issue by karlhaas** [ngrx/router#140]
Hi,

I'm using several guards for my routes. For example one to make sure that something has been fetched from the server and another in order to check if the user is authenticated.

example config:
```
  {
    path: '/something',
    component: SomethingComponent,
    guards: [EnvironmentLoadedGuard, UserAuthenticatedGuard]
  },

```

Now I got 2 questions:

1. is it possible to stop the second guard from being executed if the first guard already returns Observable.of(false)?
2. is it possible to configure a global guard which will be added automatically for all routes?

Thx.

**Comment by brandonroberts** [comment]
1. Yes. The guards are run in parallel, but if one of the guards returns false while the others are still running, they will be cancelled.
2. You could achieve this using a pathless route as your top level route and putting your routes as children of it.

```ts
[
  {
    component: Root,
    guards: [...],
    children: [
      { path: '/', component: Home },
      { path: '/away', component: Away }
    ]
 }
]
```

**Comment by karlhaas** [comment]
thx for the quick reply!

2. I tried it this way but the guards of the root component don't execute. It works when i add the same guard to the children
1. In my case both guards redirect the router in case something is missing. It happened that both called router.go. I ended up calling one guard within the other ... in my opinion this reduces the flexibility of using several guards which just have one specific check to perform.

maybe it would be nice to be able to specify if the guards run in parallel or successively

**Comment by brandonroberts** [comment]
There's nothing preventing you from creating a guard that composes the observables of two other guards together to run in them successively. Guards are just services after all so they can be injected into each other.

**Comment by karlhaas** [comment]
Yes, I already called one guard inside another which works fine. Would be nice to have it in the configuration somehow.
Thx for answering!

**Comment by brandonroberts** [comment]
No problem!
